public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +62,289 @@ const TITLE_TEXT_DICT = {
+        }
+    }
+}`;
 
-function buildPermissionOptions(optionsList: string[]) {
-  return optionsList.map(e => ({ label: e }));
+function IndexPatternRow(props: {
+  value: OptionSeletion;
+  onChangeHandler: (s: OptionSeletion) => void;
+  onCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow headerText="Index" helpText="Specify index pattern using *">
+      <EuiComboBox
+        noSuggestions
+        placeholder="Search for index name or type in index pattern"
+        selectedOptions={props.value}
+        onChange={props.onChangeHandler}
+        onCreateOption={props.onCreateHandler}
+      />
+    </FormRow>
+  );
+}
+
+function IndexPermissionRow(props: {
+  value: OptionSeletion;
+  permisionOptionsSet: OptionSeletion;
+  onChangeHandler: (s: OptionSeletion) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Index permissions"
+      headerSubText="You can specify permissions using both action groups or single permissions. 
+        An permission group is a list of single permissions.
+        You can often achieve your desired security posture using some combination of the default permission groups. 
+        You can also create your own reusable permission groups."
+    >
+      <EuiFlexGroup>
+        <EuiFlexItem style={{ maxWidth: '400px' }}>
+          <EuiComboBox
+            placeholder="Search for action group name or permission name"
+            options={props.permisionOptionsSet}
+            selectedOptions={props.value}
+            onChange={props.onChangeHandler}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton>Browse and select</EuiButton>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton iconType="popout" iconSide="right">
+            Create new permission group
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </FormRow>
+  );
+}
+
+function DocLevelSecurityRow(props: {
+  value: string;
+  onChangeHandler: (e: React.ChangeEvent<HTMLTextAreaElement>) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Document level security"
+      headerSubText="You can restrict a role to a subset of document in an index."
+      helpLink="/"
+      optional
+    >
+      <EuiTextArea
+        placeholder={FIELD_LEVEL_SECURITY_PLACEHOLDER}
+        value={props.value}
+        onChange={props.onChangeHandler}
+      />
+    </FormRow>
+  );
+}
+
+function FieldLevelSecurityRow(props: {
+  method: FieldLevelSecurityMethod;
+  fields: OptionSeletion;
+  onMethodChangeHandler: (s: string) => void;
+  onFieldChangeHandler: (s: OptionSeletion) => void;
+  onFieldCreateHandler: (s: string) => void;
+}) {
+  return (
+    <FormRow
+      headerText="Field level security"
+      headerSubText="You can restrict what document field that user can see. If you use field level security in conjunction with document-level security, make sure you don't restrict to the field that document-level security uses."
```

Comment:
Minor: per UI mockup,
what document field -> what document fields
field level -> field-level
restrict to the field -> restrict access to the fields